Author: Aaron M. Ucko <amu@ucko.debian.net>
Date:   Sun Feb 3 21:43:23 2019 -0500
Subject: Omit convenience mbed TLS copy
--- a/c++/src/connect/Makefile.connssl.lib
+++ b/c++/src/connect/Makefile.connssl.lib
@@ -2,7 +2,9 @@
 
 CPPFLAGS = $(TLS_INCLUDE) $(ORIG_CPPFLAGS)
 
-SRC_TLS  = ncbi_gnutls ncbi_mbedtls ncbi_tls \
+SRC_TLS  = ncbi_gnutls ncbi_mbedtls ncbi_tls
+
+SRC_MBEDTLS_UNUSED = \
            mbedtls/aes mbedtls/aesni mbedtls/arc4 mbedtls/asn1parse \
            mbedtls/asn1write mbedtls/base64 mbedtls/bignum mbedtls/blowfish \
            mbedtls/camellia mbedtls/ccm mbedtls/cipher mbedtls/cipher_wrap \
